lib/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,98 @@
+import React, {
+  Component,
+  ReactNode,
+  AllHTMLAttributes,
+  Fragment,
+} from 'react';
+import ThemeConsumer from '../ThemeConsumer/ThemeConsumer';
+import Box from '../Box/Box';
+import FieldLabel from '../FieldLabel/FieldLabel';
+import FieldMessage from '../FieldMessage/FieldMessage';
+import classnames from 'classnames';
+import styles from './TextField.css.js';
+
+type InputProps = AllHTMLAttributes<HTMLInputElement>;
+type RequiredInputProps = 'id' | 'value' | 'onChange';
+interface TextFieldProps
+  extends Required<Pick<InputProps, RequiredInputProps>> {
+  label?: string;
+  secondaryLabel?: ReactNode;
+  tertiaryLabel?: ReactNode;
+  placeholder?: string;
+  message?: ReactNode | false;
+  tone?: 'neutral' | 'critical' | 'positive';
+  type?: 'text' | 'password' | 'email' | 'search' | 'number' | 'tel' | 'url';
```

Comment:
As discussed, since this is a subset of allowed values, we should enforce this at runtime.